lib/WebSocket.js

```diff
@@ -848,7 +848,7 @@ function establishConnection(ReceiverClass, SenderClass, socket, upgradeHead) {
   self._receiver.onerror = function onerror(reason, errorCode) {
     // close the connection when the receiver reports a HyBi error code
     self.close(typeof errorCode !== 'undefined' ? errorCode : 1002, '');
-    self.emit('error', reason, errorCode);
+    self.emit('error', new Error(reason));
```

Comment:
@binarykitchen valid point. although I think that this function is tied closely to the implementation of the Receiver class (which obviously expects a string rather than an error object in its error interface), this is a bit of convenience for future extensions :)